src/store/reducers/currentVirtualStudy.js

```diff
@@ -47,6 +48,53 @@ const resetState = (diff = {}) => {
   return newState;
 };
 
+const constructSetSqon = (setInfo) => ({
+  op: 'and',
+  content: [
+    {
+      op: 'in',
+      content: {
+        field: 'kf_id',
+        value: `set_id:${setInfo.key}`,
+      },
+    },
+  ],
+});
+
+const addSqon = (setInfo, sqons) => {
+  const newSqon = constructSetSqon(setInfo);
+  const sqonsRemovedEmpty = sqons.filter((s) => s.content.length > 0);
+
+  return [...sqonsRemovedEmpty, newSqon];
+};
+
+const addSetSqonToSqons = (setInfo, sqons) => {
+  if (!setInfo) {
+    return { newSqons: sqons, activeIndex: 0 };
+  }
+
+  if (!sqons) {
+    return { newSqons: [{ op: 'and', content: [] }], activeIndex: 0 };
+  }
+
+  const sqonExists = sqons.some((s) =>
+    s.content.some((t) => t.content.value === `set_id:${setInfo.key}`),
+  );
+
+  if (sqonExists) {
+    return {
+      newSqons: sqons,
+      activeIndex: sqons.length - 1,
+    };
+  } else {
```

Comment:
unnecessary else, I think

Reply:
well if the set (sqon) is already in the CB we just return the unaltered sqons. If not we add it. i think it is required.